policy-controller/k8s/index/src/inbound/index.rs

```diff
@@ -1908,33 +1947,41 @@ impl<K, T> Default for NsUpdate<K, T> {
 // === impl ClusterInfo ===
 
 impl ClusterInfo {
-    fn default_inbound_http_routes<'p>(
+    fn default_inbound_routes<'p, MatchType>(
         &self,
         probe_paths: impl Iterator<Item = &'p str>,
-    ) -> HashMap<HttpRouteRef, HttpRoute> {
-        let mut routes = HashMap::with_capacity(2);
-
+    ) -> HashMap<InboundRouteRef, InboundRoute<MatchType>>
+    where
+        MatchType: GenericRouteMatch,
+        InboundRoute<MatchType>: Default,
+    {
         // If no routes are defined for the server, use a default route that
         // matches all requests. Default authorizations are instrumented on
         // the server.
-        routes.insert(HttpRouteRef::Default("default"), HttpRoute::default());
+        let mut routes = HashMap::from_iter([(
+            InboundRouteRef::DEFAULT_DEFAULT,
+            InboundRoute::<MatchType>::default(),
+        )]);
 
         // If there are no probe networks, there are no probe routes to
         // authorize.
-        if self.probe_networks.is_empty() {
+        if self.probe_networks.is_empty() || !MatchType::is::<HttpRouteMatch>() {
             return routes;
         }
 
         // Generate an `Exact` path match for each probe path defined on the
         // pod.
-        let matches: Vec<HttpRouteMatch> = probe_paths
-            .map(|path| HttpRouteMatch {
-                path: Some(PathMatch::Exact(path.to_string())),
-                headers: vec![],
-                query_params: vec![],
-                method: Some(Method::GET),
+        let matches = probe_paths
```

Comment:
we don't support automatically generating grpc probe routes.  the default inbound routes will always be http, I don't think we want this to be generic.